src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -103,6 +110,91 @@ class IpexCommunicationService extends AgentService {
     });
   }
 
+  async offerAcdc(notificationId: string) {
```

Comment:
Please check the story acceptance criteria again - I would like to be able to get a list of matching credentials so that the UI can select one, and then pass it here as the selected one.

Reply:
that's changed with I did before, right? Because I remember that now we only get the first cred record in DB to offer :'(

Reply:
Yeah I think it updated since D: We should be able to have a `getMatchingCredsForApply` and `offerAcdc(notificationId: string, acdc: ...)`

Reply:
`getMatchingCredsForApply` will be return the CredentialShortDetails, right? I guess that we will show the cred and alow user to choose it

Reply:
@bao-sotatek Good question, I think we can just return an array of the full credential details for now since it's easier and it makes it available to the UI easily.

If we need to adjust it to show short details and then fetch each time we can too but it seems wasteful since we would've already fetched all the details.

Reply:
I am thinking about pinning the acdc in CredentialShortDetail in this case, becasue now I am filtering the ACDC with signify-ts first, after that I will cast it into CredentialMetadata, and when using it to send the offer, it also need the ACDC record from signify-ts, not the CredentialMetadata

Reply:
like that 
![image](https://github.com/cardano-foundation/cf-identity-wallet/assets/142210850/e4bb13bb-65d2-4995-8a33-0e1e2561358c)

Reply:
Let me ask Robert what a UI/UX would look like for this